modules/standard/Math.chpl

```diff
@@ -1057,6 +1057,96 @@ module Math {
     return truncf(x);
   }
 
+
+  /* Returns the Bessel function of the first kind of order `0` of `x`. */
+  extern "j0f" proc j0(x: real(32)): real(32);
+
+  /* Returns the Bessel function of the first kind of order `0` of `x`. */
+  inline proc j0(x: real(64)): real(64) {
+    extern proc j0(x: real(64)): real(64);
+    return j0(x);
+  }
+
+
+  /* Returns the Bessel function of the first kind of order `1` of `x`. */
+  extern "j1f" proc j1(x: real(32)): real(32);
+
+  /* Returns the Bessel function of the first kind of order `1` of `x`. */
+  inline proc j1(x: real(64)): real(64) {
+    extern proc j1(x: real(64)): real(64);
+    return j1(x);
+  }
+
+
+  /* Returns the Bessel function of the first kind of order `n` of `x`. */
+  extern "jnf" proc jn(n: int, x: real(32)): real(32);
+
+  /* Returns the Bessel function of the first kind of order `n` of `x`. */
+  inline proc jn(n: int, x: real(64)): real(64) {
+    extern proc jn(n: c_int, x: real(64)): real(64);
+    return jn(n.safeCast(c_int), x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `0` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  extern "y0f" proc y0(x: real(32), boundChecking = true): real(32);
+
+  /* Returns the Bessel function of the second kind of order `0` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  inline proc y0(x: real(64), boundChecking = true): real(64) {
+    if boundChecking {
+      if x < 0 then
+        halt("Input value for Bessel function of second kind must be greater than 0");
+    }
+    extern proc y0(x: real(64)): real(64);
+    return y0(x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `1` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  extern "y1f" proc y1(x: real(32), boundChecking = true): real(32);
+
+  /* Returns the Bessel function of the second kind of order `1` of `x`,
+     if and only if the value of `x` is greater than 0, if 
+     `boundChecking` is `true` then for `x < 0` there will be an error,
+     otherwise an `NaN` value will be returned*/
+  inline proc y1(x: real(64), boundChecking = true): real(64) {
```

Comment:
`boundsChecking` is a global configuration param that is turned off when a Chapel program is compiled with `-sboundsChecking=true`, which is part of the `--fast` optimization flag. You can drop this argument in the procedures and rely on the global `boundsChecking` configuration param instead. We don't generally mention `boundsChecking` behavior in documentation, so you can drop mentions of `boundsChecking` in the docs as well.

Basically, when `boundsChecking` is on (which it is by default), users will get cleaner and more helpful error messages for out of bounds errors, such as `"Input value for Bessel function of second kind must be greater than 0"` in this case. When it is disabled, users lose those cleaner error messages in a trade off for performance. In this case, users will get whatever error message the C implementation spits out, when `boundsChecking` is off.

You can grep the `modules/` directory for `boundsChecking` to see other example its use.